### ISSUE TITLE
Query service to identify kinds of association data available

### ISSUE THREAD

**Issue by nlwashington** [ga4gh/schemas#349]
We need some kind of services that allow external systems to solicit any implementation for what it has in it.  For example:

1.  Does your system have diseases?
2.  Does your system have drug-response phenotypes?
3.  Does your system have pathogenicity calls?
4.  Does your system have patients?
5.  Does your system have cell lines?

As we make the associations more generic, each source will probably have only a subset of the different kinds of association data.

Basically we need a way for external systems to programmatically figure out if a given GA4GH/G2P implementation has the kind of data that's desired.  This can be implemented using either "types" of the avro object, but since the subject/object itself might be generic, a better solution is to use the types for the subject/objects themselves.  If these are typed with proper ontology terms, then they can be computed using transitive closure.
